xdsl/ir.py

```diff
@@ -911,6 +911,12 @@ def walk(self, fun: Callable[[Operation], None]) -> None:
         for op in self.ops:
             op.walk(fun)
 
+    def select_ops(self, cond: Callable[[Operation], bool]) -> List[Operation]:
```

Comment:
I'm not completely convinced that we want this in our `ir.py`.
Mostly because I would like to keep `ir.py` quite small, and I feel that
`block.select_ops(lambda x: ...)` is not much shorter than `[op for op in block.ops if ...]`